jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/ConstructorRule.java

```diff
@@ -415,4 +422,41 @@ private void generateNoArgsConstructor(JDefinedClass jclass) {
     noargsConstructor.javadoc()
         .add("No args constructor for use in serialization");
   }
+
+  private void generateVertxConstructor(JDefinedClass jclass) {
+    // add a no-args constructor for serialization purposes
+    JMethod jsonObjectConstructor = jclass.constructor(JMod.PUBLIC);
+    jsonObjectConstructor.javadoc().add("Constructor with decoding");
+    jsonObjectConstructor.param(jclass.owner().ref("io.vertx.core.json.JsonObject"), "jsonObject");
+
+    JBlock constructorBody = jsonObjectConstructor.body();
+    constructorBody.add(
+        jclass.owner()
+            .ref(jclass.fullName() + "Converter")
+            .staticInvoke("fromJson")
+            .arg(JExpr.ref("jsonObject"))
+            .arg(JExpr._this())
+    );
+  }
+
+  private void generateVertxToJsonMethod(JDefinedClass jclass) {
+    // add a no-args constructor for serialization purposes
+    JMethod toJsonMethod = jclass.method(JMod.PUBLIC, jclass.owner().ref("io.vertx.core.json.JsonObject"), "toJson");
+    toJsonMethod.javadoc().add("Encode to JsonObject");
+
+    JBlock constructorBody = toJsonMethod.body();
+    constructorBody.directStatement("JsonObject obj = new JsonObject()");
```

Comment:
It's missing a semicolon, no?

I extrapolated my own AbstractAnnotator based on your work, so thanks!

Reply:
Ops :)